### ISSUE TITLE
DateTime Axis with DiscontinuityProvider doesn't cope with empty domain

### ISSUE THREAD

**Issue by jdunkerley** [ScottLogic/d3fc#607]
If you have an empty data array the fc.util.extent function returns `[undefined, undefined]` passing this into a dateScale with a discontinuityProvider causes an error.

```javascript
var dateScale = fc.scale.dateTime()
    .domain(fc.util.extent([], 'date'))
    .discontinuityProvider(fc.scale.discontinuity.skipWeekends())
    .range([0, width]);
```

**Comment by ColinEberhardt** [comment]
Thanks for the report, what behaviour would you expect in this context?

On Tue, Oct 13, 2015 at 1:33 PM, James Dunkerley <notifications@github.com>
wrote:

> If you have an empty data array the fc.util.extent function returns [undefined,
> undefined] passing this into a dateScale with a discontinuityProvider
> causes an error.
>
> var dateScale = fc.scale.dateTime()
>     .domain(fc.util.extent([], 'date'))
>     .discontinuityProvider(fc.scale.discontinuity.skipWeekends())
>     .range([0, width]);
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ScottLogic/d3fc/issues/607>.
>



-- 
Regards,
Colin E.

**Comment by jdunkerley** [comment]
I think best to leave as undefined. 

Small change I think to the skipWeekends clampUp and clampDown functions.

So `skipWeekends.clampUp(undefined)` would return undefined.

**Comment by ColinEberhardt** [comment]
Ah, I thought you were proposing a change of behaviour for d3.util.extent!

Your proposed change makes sense and makes the code more robust.

**Comment by jdunkerley** [comment]
k will implement and PR

**Comment by chrisprice** [comment]
Fixed by #609
